### ISSUE TITLE
enhancements for repos using hooks

### ISSUE THREAD

**Issue by fluffynuts** [jesseduffield/lazygit#1109]
**Is your feature request related to a problem? Please describe.**
We use hooks on our repos, most especially at push-time, but because we're using `pre-git` (it makes life easier), even a commit can take a second or two, during which time I'm not sure if I've hit `enter` or not on the commit dialog

**Describe the solution you'd like**
short-term: the same kind of feedback that's on the Push dialog would be nice on the commit dialog (ie label and spinner)

longer-term: our push hooks can take a while. One runs the full test suite (takes around 80s) and many run a `verify-up-to-date` task which won't let you push a satellite branch if you're behind master. Both produce output, which lazygit will display if they fail (I've seen the `verify-up-to-date` output on occasion), but I'd really like a live output of commands as they are running, configured, of course, to user prefs (so the feedback dialog should be default and the details could be shown if, eg, `gui.showDetailedOperations` were set). Details should be shown _live_ though -- there's not much point in seeing the output from a long operation way after it already successfully completed, but seeing it _during_ operation provides feedback to the user

**Describe alternatives you've considered**
I wait after I _think_ I've hit enter on commit. Usually I have, but it's been a good test of holding my breath.

The last time I pushed on the project which runs tests, I exited lazygit so I could watch the tests roll by.

**Comment by jesseduffield** [comment]
I see two approaches we can take here: one is to make lazygit far better at parsing arbitrary command output. This has proven challenging (I tried it on another side project last year called (iirc) lazysession). One challenge is that your hook might open up vim or something. The other approach is to somehow see if a hook will run and switch to that hook as a subprocess. Not sure if it's actually possible to know ahead of time whether a hook will run but it should be.

**Comment by gwww** [comment]
Additional info. I've done some research on this subject. First, it would be great to capture the pre-commit output.

What I've read is don't try to be smart about it. Be dumb, like TERM=dumb type of dumb ;) One reference was a thread from tpope wrt vim-fugitive: https://github.com/tpope/vim-fugitive/issues/1446. The other reference is what to expect from a dumb terminal: https://stackoverflow.com/questions/39001517/features-obligatory-for-term-dumb-terminal

The output would go in the `Command Log` window. At least that's my guess, not being an expert `lazygit` user (yet).

This feels like an easy first cut.

As to vi or such being invoked... that's a future fix.

**Comment by jesseduffield** [comment]
@gwww I agree. I've made a start on this here: https://github.com/jesseduffield/lazygit/pull/1543
